tests/components/omie/test_sensor.py

```diff
@@ -0,0 +1,308 @@
+from freezegun import freeze_time
+from pyomie.model import OMIEResults
+import pytest
+
+from homeassistant.components.omie import OMIECoordinator
+from homeassistant.components.omie.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from .conftest import spot_price_fetcher
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+@pytest.fixture
+def mock_config_entry() -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    return MockConfigEntry(
+        title="OMIE",
+        domain=DOMAIN,
+        unique_id="omie_singleton",
+    )
+
+
+async def test_sensor_setup(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_pyomie,
+) -> None:
+    """Test sensor platform setup."""
+    mock_config_entry.add_to_hass(hass)
+
+    mock_pyomie.spot_price.side_effect = spot_price_fetcher({})
+
+    assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Check that entities were created
+    entity_registry = er.async_get(hass)
+    entities = er.async_entries_for_config_entry(
+        entity_registry, mock_config_entry.entry_id
+    )
+
+    # Should have 2 sensors (PT and ES)
+    assert len(entities) == 2
+
+    # Check entity unique IDs
+    entity_ids = {entity.unique_id for entity in entities}
+    expected_ids = {"pt_spot_price", "es_spot_price"}
+    assert entity_ids == expected_ids
+
+
+async def test_sensor_state_lisbon_timezone(
+    hass_lisbon: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_pyomie,
+    mock_omie_results_jan15,
+    mock_omie_results_jan16,
+) -> None:
+    """Test sensor state updates in Lisbon timezone across publication boundary."""
+    mock_config_entry.add_to_hass(hass_lisbon)
+
+    mock_pyomie.spot_price.side_effect = spot_price_fetcher(
+        {
+            "2024-01-15": mock_omie_results_jan15,
+            "2024-01-16": mock_omie_results_jan16,
+        }
+    )
+
+    # Step 1: 1 PM CET (before 13:30 CET publication)
+    # Lisbon day spans two CET dates: Jan 15 available, Jan 16 not yet published
+    with freeze_time("2024-01-15T12:01:00Z"):  # 12:01 UTC = 12:01 Lisbon = 13:01 CET
+        assert await hass_lisbon.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass_lisbon.async_block_till_done()
+
+        # Only 1 API call for Jan 15 (Jan 16 not yet published at 13:30 CET)
+        assert mock_pyomie.spot_price.call_count == 1
+
+    # Step 2: 3 PM CET - verify listeners update with existing data, no new API calls
+    with freeze_time("2024-01-15T14:01:00Z"):  # 14:01 UTC = 14:01 Lisbon = 15:01 CET
+        async_fire_time_changed(hass_lisbon, dt_util.utcnow())
+        await hass_lisbon.async_block_till_done()
+
+        # No additional API calls should be made
+        assert mock_pyomie.spot_price.call_count == 1
```

Comment:
```suggestion
        assert mock_pyomie.spot_price.call_count == 0
```
we could reset the call count of mock_pyomie.spot_price after the first check on L84, so here we would just compare it to zero, which makes it clearer that no calls are expected.
but feel free to keep as it is if you prefer it.